### PR TITLE
Fixes back #12, back navigation issue

### LINKED ISSUE #12: Router bug: hitting browser back button 2 or more times fails to navigate
I've found a bug in the Router where if you navigate backwards two or more times, it'll never navigate to the previous page.

[Minimal repro here](https://app-tools-router-bug.glitch.me) (and [source here](https://glitch.com/edit/#!/app-tools-router-bug?path=public%2Findex.js%3A11%3A10)).

Repro steps:
1. Navigate to page A
2. Navigate to page B
3. Navigate to page C
4. Click browser back button (result: successfully navigates to page B)
5. Click browser back button (result: bug, it navigates to page B, which we're already on)

I don't yet know the source of the bug, but I'd be glad to submit a PR if you're willing to accept it.

### PR DESCRIPTION
**Pull request by JudahGabriel**
This PR fixes #12, backward navigation bug, by skipping `pushState` when we navigate backward.

### CODE REVIEW

File: router/index.js [hunk 1]
```diff
@@ -137,9 +137,10 @@ export class Router extends EventTarget {
   }
 
   /**
-   * @param {string | URL} url 
+   * @param {string | URL} url The URL to navigate to.
+   * @param {boolean | null | undefined} isBackNav Whether the navigation is a backward navigation (e.g. clicking the browser back button.) Backward navigations won't push a new URL into the browser history.
    */
-  async navigate(url) {
+  async navigate(url, isBackNav) {
```
Comment (thepassle) [inline router/index.js]:
```suggestion
  async navigate(url, {backNav = false} = {}) {
```

File: router/index.js [hunk 2]
```diff
@@ -137,9 +137,10 @@ export class Router extends EventTarget {
   }
 
   /**
-   * @param {string | URL} url 
+   * @param {string | URL} url The URL to navigate to.
+   * @param {boolean | null | undefined} isBackNav Whether the navigation is a backward navigation (e.g. clicking the browser back button.) Backward navigations won't push a new URL into the browser history.
```
Comment (thepassle) [inline router/index.js]:
```suggestion
   * @param {{
   *  backNav?: boolean
   * }} options
```

File: router/index.js [hunk 3]
```diff
@@ -106,7 +106,7 @@ export class Router extends EventTarget {
   }
 
   _onPopState = () => {
-    this.navigate(new URL(window.location.href));
+    this.navigate(new URL(window.location.href), true);
```
Comment (thepassle) [inline router/index.js]:
```suggestion
    this.navigate(new URL(window.location.href), { backNav: true });
```

**Comment by JudahGabriel** [comment]
I've updated the code per your suggestion to use an options object with `{ backNav: true }`.

**Comment by thepassle** [comment]
Thanks for the fix and these bug reports! Super helpful :)

Do you mind if I ask what you're building? Is it on github somewhere? Would love to see it

**Comment by thepassle** [comment]
published as `0.8.9`

**Comment by JudahGabriel** [comment]
Thanks! To answer your question, I'll DM you over Twitter.

**Pull request merged**